Jenkinsfile

```diff
@@ -90,6 +85,31 @@ pipeline {
         }
       }
     }
+    /**
+      Publish Docker images.
+    */
+    stage('Publish Docker image'){
+      environment {
+        DOCKER_IMG_TAG = "${env.DOCKER_IMG}:${env.GIT_BASE_COMMIT}"
+        DOCKER_IMG_TAG_BRANCH = "${env.DOCKER_IMG}:${env.BRANCH_NAME}"
+      }
+      steps {
+        deleteDir()
+        unstash 'source'
+        dir("${BASE_DIR}"){
+          dockerLogin(secret: "${env.DOCKER_REGISTRY_SECRET}",
+            registry: "${env.DOCKER_REGISTRY}")
+          sh(label: 'Build Docker image',
+            script: "docker build -t ${env.DOCKER_IMG_TAG} .")
+          sh(label: 'Push Docker image',
+            script: "docker push ${env.DOCKER_IMG_TAG}")
+          sh(label: 'Re-tag Docker image',
+            script: "docker tag ${env.DOCKER_IMG_TAG} ${env.DOCKER_IMG_TAG_BRANCH}")
+            sh(label: 'Push Docker image',
+              script: "docker push ${env.DOCKER_IMG_TAG_BRANCH}")
```

Comment:
does the push requires a --force flag when the tag already exists?

Reply:
no, it is not needed, we push without `--force` here and it is working https://github.com/elastic/apm-pipeline-library/blob/master/.ci/buildDockerImages.groovy#L340